src/services/codefixes/inferFromUsage.ts

```diff
@@ -333,21 +333,50 @@ namespace ts.codefix {
     }
 
     function annotateJSDocParameters(changes: textChanges.ChangeTracker, sourceFile: SourceFile, parameterInferences: readonly ParameterInference[], program: Program, host: LanguageServiceHost): void {
-        const signature = parameterInferences.length && parameterInferences[0].declaration.parent;
+        const signature = parameterInferences.length && first(parameterInferences).declaration.parent;
         if (!signature) {
             return;
         }
-        const paramTags = mapDefined(parameterInferences, inference => {
+
+        const inferences = mapDefined(parameterInferences, inference => {
             const param = inference.declaration;
             // only infer parameters that have (1) no type and (2) an accessible inferred type
-            if (param.initializer || getJSDocType(param) || !isIdentifier(param.name)) return;
-
+            if (param.initializer || getJSDocType(param) || !isIdentifier(param.name)) {
+                return;
+            }
             const typeNode = inference.type && getTypeNodeIfAccessible(inference.type, param, program, host);
-            const name = factory.cloneNode(param.name);
-            setEmitFlags(name, EmitFlags.NoComments | EmitFlags.NoNestedComments);
-            return typeNode && factory.createJSDocParameterTag(/*tagName*/ undefined, name, /*isBracketed*/ !!inference.isOptional, factory.createJSDocTypeExpression(typeNode), /* isNameFirst */ false, "");
+            if (typeNode) {
+                const name = factory.cloneNode(param.name);
+                setEmitFlags(name, EmitFlags.NoComments | EmitFlags.NoNestedComments);
+                return { name: factory.cloneNode(param.name), param, isOptional: !!inference.isOptional, typeNode };
+            }
         });
-        addJSDocTags(changes, sourceFile, signature, paramTags);
+
+        if (!inferences.length) {
+            return;
+        }
+
+        if ((isArrowFunction(signature) || isFunctionExpression(signature)) && isCallExpression(signature.parent)) {

```

Comment:
Doesn't this need to happen everywhere except the initialiser of a variable declaration? That might not quite be right, but I expected to see some negated condition instead of `isCallExpression`.

Reply:
Based on the issue description, I thought that we need to use this kind of type of definition, to arrow functions or function expressions that used as arguments in call expressions. Like so

```ts
foo((a) => {})
```

```ts
foo(function (a) => {})
```

Reply:
I think the codefix will also generate incorrect code for the following case, for example:

```ts
/** @type {} */
let fs = [x => x + 1, y => y + 2]
```

Though perhaps the codefix wouldn't even apply with or without the `@type` annotation.